java/src/main/java/ai/rapids/cudf/Cuda.java

```diff
@@ -269,18 +269,30 @@ public void close() {
    * @param kind  - Type of transfer. {@link CudaMemcpyKind}
    */
   static void memcpy(long dst, long src, long count, CudaMemcpyKind kind) {
-    memcpy(dst, src, count, kind.getValue());
+    memcpy(dst, src, count, kind, DEFAULT_STREAM);
   }
 
-  private static native void memcpy(long dst, long src, long count, int kind) throws CudaException;
+  /**
+   * Copies count bytes from the memory area pointed to by src to the memory area pointed to by
+   * dst.
+   * Calling cudaMemcpy() with dst and src pointers that do not
+   * match the direction of the copy results in an undefined behavior.
+   * @param dst   - Destination memory address
+   * @param src   - Source memory address
+   * @param count - Size in bytes to copy
+   * @param kind  - Type of transfer. {@link CudaMemcpyKind}
+   */
+  static void asyncMemcpy(long dst, long src, long count, CudaMemcpyKind kind) {
+    asyncMemcpy(dst, src, count, kind, DEFAULT_STREAM);
+  }
 
   /**
    * Sets count bytes starting at the memory area pointed to by dst, with value.
    * @param dst   - Destination memory address
    * @param value - Byte value to set dst with
    * @param count - Size in bytes to set
    */
-  public static native void memset(long dst, byte value, long count) throws CudaException;
```

Comment:
This method is a public API and a bit painful for downstream projects if removed.   Should it be preserved and implement it in terms of `memcpyAsync` and `cudaStreamSynchronize` as done for the existing `memcpy` method?

Reply:
Done.